src/main/java/com/stuypulse/robot/subsystems/ColorSensor.java

```diff
@@ -9,55 +9,140 @@
 
 import com.stuypulse.robot.Constants.Colors;
 
+import edu.wpi.first.wpilibj.DriverStation;
 import edu.wpi.first.wpilibj.I2C.Port;
 import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
 import edu.wpi.first.wpilibj.util.Color;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
 public class ColorSensor extends SubsystemBase {
 
+    // Woof Color enum 
+    public static enum WColor {
+        // Create all colors on the control panel
+        YELLOW,
+        CYAN,
+        GREEN,
+        RED,
+        
+        // To avoid null pointer exceptions when returning error states
+        NONE;
+
+        // Each color will store information about its position
+        private WColor next, rotated;
+
+        // Use a static block to setup circular references
+        static {
+            /*
+            Control Panel Reference: 
+            https://firstfrc.blob.core.windows.net/frc2021/Manual/Sections/2021FRCGameManualSection03.pdf
+            */
+
+            // Setup Yellow
+            YELLOW.next = CYAN; 
+            YELLOW.rotated = GREEN;
+            
+            // Setup Cyan
+            CYAN.next = GREEN;
+            CYAN.rotated = RED;
+            
+            // Setup Green
+            GREEN.next = RED;
+            GREEN.rotated = YELLOW;
+
+            // Setup Red
+            RED.next = YELLOW;
+            RED.rotated = CYAN; 
+
+            // Setup Error state
+            NONE.next = NONE;
+            NONE.rotated = NONE;
+        }
+
+        // Getters 
+        public WColor getNextColor() {
+            return this.next;
+        }
+
+        public WColor getRotatedColor() {
+            return this.rotated;
```

Comment:
change this to `this.next.next`

Reply:
Thanks for pointing this out, I changed it